jetty-security/src/main/java/org/eclipse/jetty/security/SecurityHandler.java

```diff
@@ -58,20 +61,34 @@
 public abstract class SecurityHandler extends HandlerWrapper implements Authenticator.AuthConfiguration
 {
     private static final Logger LOG = Log.getLogger(SecurityHandler.class);
+    private static final List<Authenticator.Factory> _authenticatorFactories = new ArrayList<>();
 
     private boolean _checkWelcomeFiles = false;
     private Authenticator _authenticator;
-    private Authenticator.Factory _authenticatorFactory = new DefaultAuthenticatorFactory();
+    private Authenticator.Factory _authenticatorFactory;
     private String _realmName;
     private String _authMethod;
-    private final Map<String, String> _initParameters = new HashMap<String, String>();
+    private final Map<String, String> _initParameters = new HashMap<>();
     private LoginService _loginService;
     private IdentityService _identityService;
     private boolean _renewSession = true;
 
+    static
+    {
+        for (Authenticator.Factory factory : ServiceLoader.load(Authenticator.Factory.class))
+        {
+            _authenticatorFactories.add(factory);
+        }
+
+        _authenticatorFactories.add(new DefaultAuthenticatorFactory());
+    }
+
     protected SecurityHandler()
     {
-        addBean(_authenticatorFactory);
+        for (Authenticator.Factory factory : _authenticatorFactories)
+        {
+            addBean(factory);
```

Comment:
The known factories should not be beans.... at least not like this.
Perhaps the whole collection can be included in a dump as a named collection called "knownFactories"